Code/GraphMol/Subset.cpp

```diff
@@ -0,0 +1,346 @@
+//
+//  Copyright (C) 2025 Hussein Faara, Brian Kelley and other RDKit Contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/GraphMol.h>
+#include <GraphMol/QueryAtom.h>
+#include <GraphMol/QueryBond.h>
+#include "Subset.h"
+
+namespace RDKit {
+namespace {
+
+inline void copyComputedProps(const ROMol &src, ROMol &dst) {
+  dst.updateProps(src);
+  for(auto &v : dst.getPropList(true, false)) {
+    if(v != RDKit::detail::computedPropName)
+      dst.clearProp(v);
+  }
+}
+
+static void copySelectedAtomsAndBonds(::RDKit::RWMol &extracted_mol,
```

Comment:
Since we're inside the `RDKit` namespace, I don't think it needs to be used anywhere in this code.